diff --git a/pom.xml b/pom.xml
--- a/pom.xml
+++ b/pom.xml
@@ -192,8 +192,8 @@
     <jffi.version>1.2.9</jffi.version>
 
     <jline.version>2.13</jline.version>
-    <lucene.version>5.1.0_1</lucene.version>
-    <lucene.osgi-version>5.1.0.1</lucene.osgi-version>
+    <lucene.version>4.7.0_1</lucene.version>
+    <lucene.osgi-version>4.7.0.1</lucene.osgi-version>
     <lucene-analysis.bundle.symbolic-name>org.apache.servicemix.bundles.lucene-analyzers-common</lucene-analysis.bundle.symbolic-name>
     <lucene-core.bundle.symbolic-name>org.apache.servicemix.bundles.lucene</lucene-core.bundle.symbolic-name>
     <!--
diff --git a/vm-heap-analysis/common/src/main/java/com/redhat/thermostat/vm/heap/analysis/common/HeapDump.java b/vm-heap-analysis/common/src/main/java/com/redhat/thermostat/vm/heap/analysis/common/HeapDump.java
--- a/vm-heap-analysis/common/src/main/java/com/redhat/thermostat/vm/heap/analysis/common/HeapDump.java
+++ b/vm-heap-analysis/common/src/main/java/com/redhat/thermostat/vm/heap/analysis/common/HeapDump.java
@@ -66,6 +66,7 @@
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.LockObtainFailedException;
 import org.apache.lucene.store.RAMDirectory;
+import org.apache.lucene.util.Version;
 
 import com.redhat.thermostat.common.utils.LoggingUtils;
 import com.redhat.thermostat.vm.heap.analysis.common.model.HeapInfo;
@@ -140,7 +141,7 @@
 
         Enumeration<JavaHeapObject> thingos = snapshot.getThings();
         Directory dir = new RAMDirectory();
-        IndexWriterConfig indexWriterConfig = new IndexWriterConfig(new SimpleAnalyzer());
+        IndexWriterConfig indexWriterConfig = new IndexWriterConfig(Version.LUCENE_47, new SimpleAnalyzer(Version.LUCENE_47));
         IndexWriter writer = new IndexWriter(dir, indexWriterConfig);
         while (thingos.hasMoreElements()) {
             JavaHeapObject thingo = thingos.nextElement();
